### PR TITLE
Rewrite clear and others

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #5742 

Depends on
- #5763

Interval-major rewrites of functions using WaveTrack::HandleClear, other than ClearAndPaste

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: libraries/lib-wave-track/WaveTrack.cpp [hunk 1]
```diff
@@ -272,14 +272,14 @@ void WaveTrack::Interval::SetTrimRight(double t)
       GetClip(channel)->SetTrimRight(t);
 }
 
-void WaveClip::Interval::TrimLeft(double deltaTime)
+void WaveTrack::Interval::TrimLeft(double deltaTime)
```
Comment (saintmatthieu) [inline libraries/lib-wave-track/WaveTrack.cpp:275]: ... oh, there we go. You might want to squash these with the previous commit, which probably won't compile.

**Review by saintmatthieu — APPROVED** [review]
Clean-up looks good and sequence of commit made the review almost entertaining :D

File: libraries/lib-wave-track/WaveTrack.cpp (hunk 1 above)
Comment (Paul-Licameli) [reply libraries/lib-wave-track/WaveTrack.cpp:275]: Yes

**Comment by Paul-Licameli** [comment]
Rebased unchanged just to clean up the commit history

**Review by saintmatthieu — APPROVED** [review]

**Comment by Paul-Licameli** [comment]
Rebased without change onto the updated base branch

**Pull request merged**